ta4j-core/src/main/java/org/ta4j/core/analysis/criteria/BuyAndHoldCriterion.java

```diff
@@ -35,7 +35,13 @@ this software and associated documentation files (the "Software"), to deal in
 
     @Override
     public double calculate(TimeSeries series, TradingRecord tradingRecord) {
-        return series.getBar(series.getEndIndex()).getClosePrice().dividedBy(series.getBar(series.getBeginIndex()).getClosePrice()).doubleValue();
+        if (tradingRecord.getTradeCount() == 0) {
+            return 1.0d;
+        }
+        int firstEntryIndex = tradingRecord.getTrades().get(0).getEntry().getIndex();
+        int lastExitIndex = tradingRecord.getTrades().get(tradingRecord.getTradeCount()-1).getExit().getIndex();
+
+        return series.getBar(lastExitIndex).getClosePrice().dividedBy(series.getBar(firstEntryIndex).getClosePrice()).doubleValue();
```

Comment:
Is it correct to say that the range of the `TradingRecord` is just from first entry to last exit index of its trades? Maybe the run was over a much bigger range, but the first trade started after some bars or the last exit was earlier. In my opinion in most cases this assumption is not valid, the `TradingRecord` needs start and end indices that are set in TimeSeries#run()